### PR TITLE
chore: add shfmt

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
Given all the shell files that now exist in the repo, would be nice to have linting/formatting around them (in addition to the existing shellcheck which doesn't do anything to format the shell code). This PR introduces `shfmt` to both check for changes and apply formatting when the associated make targets are called.

### CODE REVIEW

File: examples/ingest/pinecone/ingest.sh [hunk 1]
```diff
@@ -5,26 +5,25 @@
...
-        --index-name "<Pinecone index name, ie: ingest-test>" \
-        --environment "<Pinecone index name, ie: ingest-test>" \
-        --batch-size "<Number of elements to be uploaded per batch, ie. 80>" \
-        --num-processes "<Number of processes to be used to upload, ie. 2>"
+	local \
+	--input-path example-docs/book-war-and-peace-1225p.txt \
+	--output-dir local-to-pinecone \
+	--strategy fast \
+	--chunk-elements \
+	--embedding-provider unstructured embedding provider, ie. langchain-huggingface \
```
Comment (potter-potter) [inline examples/ingest/pinecone/ingest.sh:20]: This looks wrong

File: examples/ingest/pinecone/ingest.sh [hunk 2]
```diff
@@ -5,26 +5,25 @@
...
+	2 \
```
Comment (potter-potter) [inline examples/ingest/pinecone/ingest.sh:21]: And  here looks wrong too

File: scripts/docker-build.sh [hunk 3]
```diff
@@ -5,16 +5,16 @@ DOCKER_REPOSITORY="${DOCKER_REPOSITORY:-quay.io/unstructured-io/unstructured}"
 PIP_VERSION="${PIP_VERSION:-23.1.2}"
 DOCKER_IMAGE="${DOCKER_IMAGE:-unstructured:dev}"
 
-DOCKER_BUILD_CMD=(docker buildx build --load -f Dockerfile \
-  --build-arg PIP_VERSION="$PIP_VERSION" \
-  --build-arg BUILDKIT_INLINE_CACHE=1 \
-  --progress plain \
-  --cache-from "$DOCKER_REPOSITORY":latest \
-  -t "$DOCKER_IMAGE" .)
+DOCKER_BUILD_CMD=(docker buildx build --load -f Dockerfile
```
Comment (potter-potter) [inline scripts/docker-build.sh:8]: Double check that backslashes aren't needed. :)

Comment (rbiseck3) [reply scripts/docker-build.sh:8]: Verified, this works when building the docker image

File: scripts/performance/benchmark.sh [hunk 4]
```diff
@@ -20,49 +20,48 @@ S3_RESULTS_DIR="performance-test/results"
...
+		unstructured:perf-test \
+		bash /home/notebook-user/scripts/performance/benchmark-local.sh 2>&1 | tee >(while IFS= read -r line; do
+			read_benchmark_logs_for_results
+		done)
 else
-    NUM_ITERATIONS="$NUM_ITERATIONS" INSTANCE_TYPE="$INSTANCE_TYPE" GIT_HASH="$GIT_HASH" SLOW_FILES="${SLOW_FILES[*]}" HI_RES_STRATEGY_FILES="${HI_RES_STRATEGY_FILES[*]}" "$SCRIPT_DIR"/benchmark-local.sh 2>&1 | \
-    tee >(while IFS= read -r line; do
-        read_benchmark_logs_for_results
-    done)
+	NUM_ITERATIONS="$NUM_ITERATIONS" INSTANCE_TYPE="$INSTANCE_TYPE" GIT_HASH="$GIT_HASH" SLOW_FILES="${SLOW_FILES[*]}" HI_RES_STRATEGY_FILES="${HI_RES_STRATEGY_FILES[*]}" "$SCRIPT_DIR"/benchmark-local.sh 2>&1 |
```
Comment (potter-potter) [inline scripts/performance/benchmark.sh:52]: I don't know shell well enough to tell if the backslash can be removed...

**Comment by potter-potter** [comment]
This looks good. We are going to have some good looking shell scripts! Mostly it was lots of simple reformatting. I was a little hesitant when it removed some backslashes. But that was it.

File: scripts/performance/benchmark.sh (hunk 4 above)
Comment (rbiseck3) [reply scripts/performance/benchmark.sh:52]: verified this works

File: examples/ingest/pinecone/ingest.sh (hunk 2 above)
Comment (potter-potter) [reply examples/ingest/pinecone/ingest.sh]: This should still be --num-processes 2 \

File: examples/ingest/pinecone/ingest.sh (hunk 1 above)
Comment (potter-potter) [inline examples/ingest/pinecone/ingest.sh]: This should have quotations and then "<an blablabla>"

**Review by potter-potter — APPROVED** [review]
Looks great!

**Comment by cragwolfe** [comment]
The massive indentation in some scripts feels less reader-friendly to me. Can some scripts go back to 2 spaces per level of indentation?

E.g., https://github.com/Unstructured-IO/unstructured/blob/main/scripts/user/unstructured-get-json.sh

**Pull request merged**